functions/src/import-votes.ts

```diff
@@ -0,0 +1,50 @@
+import * as functions from "firebase-functions";
+import * as firebase from "firebase-admin";
+
+import { Config } from "./config";
+import { Vote } from "./cast-vote";
+
+const config = functions.config() as Config;
+const importVotesConfigs = config.features.import_votes;
+const db = firebase.firestore();
+
+const fromJSONtoVote = (JSONString: string) => {
+  const supposedlyValidVotes = JSON.parse(JSONString);
+  const validVotes: Vote[] = supposedlyValidVotes.votes.filter(
+    supposedlyValidVote =>
+      supposedlyValidVote.value &&
+      supposedlyValidVote.campaign &&
+      supposedlyValidVote.agency
+  );
+  return validVotes;
+};
+
+export const importVotes = functions.https.onRequest(
+  async (req: functions.Request, res: functions.Response) => {
+    if (!importVotesConfigs.enabled) {
+      console.info("Feature import votes disabled, aborting");
+      return;
+    }
+    const passedAuthData = req.get("Authorization") || "";
+    const passedAuthKey = passedAuthData === `Bearer ${importVotesConfigs.key}`;
+    if (!passedAuthKey) {
+      console.info("Passed the wrong auth key, aborting");
+      return;
```

Comment:
This won't trigger any error and the function will timeout, use res code 401 instead